nav2_simple_navigator/src/simple_navigator.cpp

```diff
@@ -26,26 +25,89 @@ namespace nav2_simple_navigator
 {
 
 SimpleNavigator::SimpleNavigator()
-: Node("SimpleNavigator")
+: nav2_lifecycle::LifecycleNode("simple_navigator")
+{
+  RCLCPP_INFO(get_logger(), "Creating");
+}
+
+SimpleNavigator::~SimpleNavigator()
 {
-  RCLCPP_INFO(get_logger(), "Initializing");
+  RCLCPP_INFO(get_logger(), "Destroying");
+}
 
-  auto temp_node = std::shared_ptr<rclcpp::Node>(this, [](auto) {});
+nav2_lifecycle::CallbackReturn
+SimpleNavigator::on_configure(const rclcpp_lifecycle::State & state)
+{
+  RCLCPP_INFO(get_logger(), "on_configure");
 
-  planner_client_ =
-    std::make_unique<nav2_tasks::ComputePathToPoseTaskClient>(temp_node);
+  auto node = shared_from_this();
 
-  controller_client_ = std::make_unique<nav2_tasks::FollowPathTaskClient>(temp_node);
+  planner_client_ = std::make_unique<nav2_tasks::ComputePathToPoseTaskClient>(node);
+  controller_client_ = std::make_unique<nav2_tasks::FollowPathTaskClient>(node);
+  task_server_ = std::make_unique<nav2_tasks::NavigateToPoseTaskServer>(node);
 
-  task_server_ = std::make_unique<nav2_tasks::NavigateToPoseTaskServer>(temp_node);
+  task_server_->on_configure(state);
+  planner_client_->on_configure(state);
+  controller_client_->on_configure(state);
 
   task_server_->setExecuteCallback(
     std::bind(&SimpleNavigator::navigateToPose, this, std::placeholders::_1));
+
+  return nav2_lifecycle::CallbackReturn::SUCCESS;
 }
 
-SimpleNavigator::~SimpleNavigator()
+nav2_lifecycle::CallbackReturn
+SimpleNavigator::on_activate(const rclcpp_lifecycle::State & state)
+{
+  RCLCPP_INFO(get_logger(), "on_activate");
+
+  task_server_->on_activate(state);
+  planner_client_->on_activate(state);
+  controller_client_->on_activate(state);
+
+  return nav2_lifecycle::CallbackReturn::SUCCESS;
+}
+
+nav2_lifecycle::CallbackReturn
+SimpleNavigator::on_deactivate(const rclcpp_lifecycle::State & state)
+{
+  RCLCPP_INFO(get_logger(), "on_deactivate");
+
+  task_server_->on_deactivate(state);
+  planner_client_->on_deactivate(state);
+  controller_client_->on_deactivate(state);
+
+  return nav2_lifecycle::CallbackReturn::SUCCESS;
+}
+
+nav2_lifecycle::CallbackReturn
+SimpleNavigator::on_cleanup(const rclcpp_lifecycle::State & state)
+{
+  RCLCPP_INFO(get_logger(), "on_cleanup");
+
+  task_server_->on_cleanup(state);
+  planner_client_->on_cleanup(state);
+  controller_client_->on_cleanup(state);
+
+  task_server_.reset();
+  planner_client_.reset();
+  controller_client_.reset();
+
+  return nav2_lifecycle::CallbackReturn::SUCCESS;
+}
+
+nav2_lifecycle::CallbackReturn
+SimpleNavigator::on_error(const rclcpp_lifecycle::State &)
+{
+  RCLCPP_INFO(get_logger(), "on_error");
```

Comment:
Shouldn't this print an RCLCPP_ERROR message of some kind?

Reply:
Yes, good point. I'll update these.